docs/examples.rst

```diff
@@ -461,6 +461,51 @@ The metadata dictionary follows the normal dictionary rules: keys need to be has
 If you're the author of a third-party library with ``attrs`` integration, please see :ref:`Extending Metadata <extending_metadata>`.
 
 
+Types
+-----
+
+``attrs`` also allows you to associate a type with an attribute using either the *type* argument to :func:`attr.ib` or -- as of Python 3.6 -- using `PEP 526 <https://www.python.org/dev/peps/pep-0526/>`_-annotations:
+
+
+.. doctest::
+
+   >>> @attr.s
+   ... class C:
+   ...     x = attr.ib(type=int)
+   ...     y: int = attr.ib()
+   >>> attr.fields(C).x.type
+   <class 'int'>
+   >>> attr.fields(C).y.type
+   <class 'int'>
+
+If you don't mind annotating all attributes, you can even drop the :func:`attr.ib`:
+
+.. doctest::
+
+   >>> import typing
+   >>> @attr.s(auto_attribs=True)
+   ... class AutoC:
+   ...     x: int
+   ...     y: int
+   ...     foo: typing.Any = attr.ib(
+   ...          default="every attrib needs a type if auto_attribs=True"
+   ...     )
```

Comment:
I'd consider making this a `str`, since that's the type of the default, and absent some reason not given here, it just looks like a bad practice to use `Any`.